braindecode/datasets/dataset.py

```diff
@@ -0,0 +1,71 @@
+"""
+Dataset classes.
+"""
+
+# Authors: Hubert Banville <hubert.jbanville@gmail.com>
+#          Lukas Gemein <l.gemein@gmail.com>
+#          Simon Brandt <simonbrandt@protonmail.com>
+#          David Sabbagh <dav.sabbagh@gmail.com>
+#
+# License: BSD (3-clause)
+
+import numpy as np
+from torch.utils.data import Dataset
+
+
+class WindowsDataset(Dataset):
+    """
+    Dataset wrapper around mne.Epochs
+
+    Parameters
+    ----------
+    windows : mne.Epochs
+        epoched data to wrap
+    target : str
+        target specified by user to be decoded
+    transforms : transform object | list of transform objects | None
+        preprocessor transform(s) applied sequentially on windowed data.
+    """
+
+    def __init__(self, windows, target="target", transforms=None):
+        self.windows = windows
+        self.target = target
+        if self.target != "target":
+            assert self.target in self.windows.info["subject_info"].keys()
+        self.transforms = (
+            [transforms] if not isinstance(transforms, list) else transforms
+        )
+
+        # XXX Handle multitarget case
+
+    def __getitem__(self, index):
+        """Return one window and its label
+
+        Parameters
+        ----------
+        index : int
+            index of the window to return
+
+        Returns
+        -------
+        x : ndarray, shape (n_channels, window_size)
+            one window of data
+        y : int | float
+            window target
+        """
+        x = np.squeeze(self.windows[index].get_data())
```

Comment:
I recommend to avoid using squeeze by any mean. Here you know what index you want to kill so just do
```suggestion
        x = self.windows[index].get_data()[0]
```

Reply:
also that this will typically be unecessarily slow as indexing epochs constructs a new epochs and makes a copy of windows.info which will take time and not needed here. I would consider adding a parameter to get_data in mne to avoid such copy